src/app/admin/packingSlotsTable/PackingSlotsTable.tsx

```diff
@@ -0,0 +1,296 @@
+            .then((response) => setRows(response))
+            .catch((error) => console.log(error))
+            .finally(() => setIsLoading(false));
+    }
+    useEffect(() => {
+        getPackingSlots();
+    }, []);
+
+    useEffect(() => {
+        // This requires that the DB table has Realtime turned on
+        const subscriptionChannel = supabase
+            .channel("packing-slot-table-changes")
+            .on(
+                "postgres_changes",
+                { event: "*", schema: "public", table: "packing_slots" },
+                async () => setRows(await fetchPackingSlots())
+            )
+            .subscribe();
+
+        return () => {
+            supabase.removeChannel(subscriptionChannel);
+        };
+    }, []);
+
+    const handleSaveClick = (id: GridRowId) => () => {
+        setRowModesModel({ ...rowModesModel, [id]: { mode: GridRowModes.View } });
+    };
+
+    const processRowUpdate = (newRow: PackingSlotRow): PackingSlotRow => {
+        setIsLoading(true);
+        if (newRow.isNew) {
+            createPackingSlot(newRow).then(() => setIsLoading(false));
+        } else {
+            updatePackingSlot(newRow).then(() => setIsLoading(false));
+        }
+        return newRow;
+    };
+
+    const handleRowModesModelChange = (newRowModesModel: GridRowModesModel): void => {
+        setRowModesModel(newRowModesModel);
+    };
+
+    const handleRowEditStop: GridEventListener<"rowEditStop"> = (params, event) => {
+        if (params.reason === GridRowEditStopReasons.rowFocusOut) {
+            event.defaultMuiPrevented = true;
+        }
+    };
+
+    const handleEditClick = (id: GridRowId) => () => {
+        setRowModesModel({ ...rowModesModel, [id]: { mode: GridRowModes.Edit } });
+    };
+
+    const handleDeleteClick = (id: GridRowId) => () => {
+        setIsLoading(true);
+        deletePackingSlot(id)
+            .then(() => setIsLoading(false))
+            .catch((error) => console.log(error))
+            .then(() => setIsLoading(false));
+    };
+
+    const handleCancelClick = (id: GridRowId) => () => {
+        setRowModesModel({
+            ...rowModesModel,
+            [id]: { mode: GridRowModes.View, ignoreModifications: true },
+        });
+
+        if (rows) {
+            const editedRow = rows.find((row) => row.id === id);
+            if (editedRow!.isNew) {
+                setRows(rows.filter((row) => row.id !== id));
+            }
+        }
+    };
+
+    const handleUpClick = (id: GridRowId, row: PackingSlotRow) => () => {
+        const rowIndex = row.order - 1;
+        if (rowIndex > 0) {
+            if (rows) {
+                setIsLoading(true);
+                const rowOne = rows[rowIndex];
+                const rowTwo = rows[rowIndex - 1];
+                swapRows(rowOne, rowTwo)
+                    .then(() => setIsLoading(false))
+                    .catch((error) => console.log(error))
+                    .finally(() => setIsLoading(false));
```

Comment:
Probably no need to set isloading to false in `then` and `finally` here because `finally` will be called regardless of the result. (same in some other places)
```suggestion
                    .catch((error) => console.log(error))
                    .finally(() => setIsLoading(false));
```